src/main/java/org/cactoos/http/HtWire.java

```diff
@@ -80,16 +86,40 @@ public HtWire(final String addr) {
      * @param tcp The TCP port
      */
     public HtWire(final String addr, final int tcp) {
+        this(addr, tcp, HtWire::socket);
+    }
+
+    /**
+     * Ctor.
+     * @param uri The address of the server
+     * @param spplier Socket supplier
+     */
+    HtWire(final URI uri,
+        final BiFunction<String, Integer, Socket> spplier) {
+        this(uri.getHost(), HtWire.prt(uri), spplier);
+    }
+
+    /**
+     * Ctor.
+     * @param addr The address of the server
+     * @param tcp The TCP port
+     * @param spplier Supplier of sockets
+     */
+    HtWire(final String addr, final int tcp,
+        final BiFunction<String, Integer, Socket> spplier) {
         this.address = addr;
         this.port = tcp;
+        this.supplier = spplier;
     }
 
     @Override
     public Input send(final Input input) throws IOException {
-        try (final Socket socket = new Socket(this.address, this.port);
+        try (
+            final Socket socket = this.supplier.apply(this.address, this.port);
             final InputStream source = input.stream();
             final InputStream ins = socket.getInputStream();
-            final OutputStream ous = socket.getOutputStream()) {
+            final OutputStream ous = socket.getOutputStream()
+        ) {
```

Comment:
@krzyk Do we really need this unnecessary refactoring here?

Reply:
@SharplEr Well, yes, it wasn't formatted according to http://www.yegor256.com/2014/10/23/paired-brackets-notation.html

Reply:
@krzyk okay then.